sphinx_gallery/tests/test_gen_rst.py

```diff
@@ -88,3 +92,36 @@ def test_extract_intro():
             result,
             'This is the description of the example which goes on and on')
         assert_false('second paragraph' in result)
+
+
+def test_pattern_matching():
+    """Test if only examples matching pattern are executed"""
+    examples_dir = tempfile.mkdtemp()
+    gallery_dir = tempfile.mkdtemp()
+
+    gallery_conf = {
+        'fname_pattern': '%s/plot_0*' % examples_dir,
+        'examples_dirs': examples_dir,
```

Comment:
There are intermediate files which are generated -- so I think it's cleaner to use a `tmp` dir rather than the actual example dir, no?